File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/actions/fight/FightTriggeredEffect.scala





















// Generated on 05/03/2013 10:35:02
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait FightTriggeredEffect extends AbstractFightDispellableEffect {
  override val typeId = 210

  protected trait Type extends AbstractFightDispellableEffect.Type with Serializable {
    override val typeId = FightTriggeredEffect.typeId

    val arg1: Int
    val arg2: Int
    val arg3: Int
    val delay: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(arg1)
      buf.writeInt(arg2)
      buf.writeInt(arg3)
      buf.writeShort(delay)
    }

  }

  protected def create(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int, arg1: Int, arg2: Int, arg3: Int, delay: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val uid = buf.readInt()
    val targetId = buf.readInt()
    val turnDuration = buf.readShort()
    val dispelable = buf.readByte()
    val spellId = buf.readShort()
    val parentBoostUid = buf.readInt()
    val arg1 = buf.readInt()
    val arg2 = buf.readInt()
    val arg3 = buf.readInt()
    val delay = buf.readShort()

    create(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid, arg1, arg2, arg3, delay)
  }

}

object FightTriggeredEffect extends FightTriggeredEffect {
  case class TypeImpl(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int, arg1: Int, arg2: Int, arg3: Int, delay: Short) extends Type
  override def create(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int, arg1: Int, arg2: Int, arg3: Int, delay: Short) = TypeImpl(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid, arg1, arg2, arg3, delay)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(uid: Int, targetId: Int, turnDuration: Short, dispelable: Byte, spellId: Short, parentBoostUid: Int, arg1: Int, arg2: Int, arg3: Int, delay: Short) = create(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid, arg1, arg2, arg3, delay)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/FightTriggeredEffect.java


// Generated on 05/08/2013 19:38:00
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class FightTriggeredEffect extends AbstractFightDispellableEffect {
    public static final short TYPE_ID = 210;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public int arg1;
    public int arg2;
    public int arg3;
    public short delay;
    
    public FightTriggeredEffect() { }
    
    public FightTriggeredEffect(int uid, int targetId, short turnDuration, byte dispelable, short spellId, int parentBoostUid, int arg1, int arg2, int arg3, short delay) {
        super(uid, targetId, turnDuration, dispelable, spellId, parentBoostUid);
        this.arg1 = arg1;
        this.arg2 = arg2;
        this.arg3 = arg3;
        this.delay = delay;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(arg1);
        buf.writeInt(arg2);
        buf.writeInt(arg3);
        buf.writeShort(delay);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        arg1 = buf.readInt();
        arg2 = buf.readInt();
        arg3 = buf.readInt();
        delay = buf.readShort();
    }
    
}
